### ISSUE TITLE
Skip Initial Entry Animation when Hydration Enabled

### ISSUE THREAD

**Issue by Char2sGu** [angular/angular#58995]
### Which @angular/* package(s) are relevant/related to the feature request?

animations

### Description

When hydration is enabled, the initial entry animation will not be played until hydration is complete, resulting in a very wierd visual effect, where the elements appear at their final position before they are animated.

### Proposed solution

Provide an option to skip the initial entry animations when hydration is enabled. A possible syntax might be:
```ts
provideClientHydration(withInitialAnimationSkipped())
```

### Alternatives considered

Alternatively, document this behavior as expected at angular.dev and provide viablae solutions.
See below for an example workaround:
```ts
@Component({
  host: { '[@.disabled]': '!stable()' },
})
export class AppComponent {
  #router = inject(Router);

  stable = toSignal(
    this.#router.events.pipe(
      pickType(NavigationEnd),
      first(),
      switchMap(() => new Promise((r) => setTimeout(r))),
      map(() => true),
      startWith(false),
    ),
    { requireSync: true },
  );
}

```

**Comment by Char2sGu** [comment]
Hey It's quite troublesome to create a SSR Angular application on Stackblitz. I'll create a dedicated repo for reproduction.

**Comment by Char2sGu** [comment]
The below video demonstrates the issue.

When the page is refreshed, the nav entry animation is delayed because of hydration, resulting in an unpleasant visual experience.

https://github.com/user-attachments/assets/2c583c43-85eb-4f7e-9ca3-1bde376d114d

The idea of entry animations is essentially in conflict with SSR, but currently there is no easy way to disable such entry animations.

**Comment by thePunderWoman** [comment]
@Char2sGu This delay is inevitable with hydration since the code doesn't exist until hydration finishes. Have you considered switching to native CSS animations for this behavior instead? That would entirely eliminate this problem.

**Comment by Char2sGu** [comment]
@thePunderWoman Exactly. This delay is inevitable in any SSR applications, and thus any initial entry animations powered by `@angular/animations` will result in a weird visual effect.

CSS animations can indeed address this issue, but for animations powered by `@angular/animations`, I feel like it makes sense to skip such entry animations (with an option or perhaps by default), since I don't think any developer would expect this behavior.

Alternatively, I would suggest a warning printed to the console in this case, with a link to solutions at angular.dev.

**Comment by Char2sGu** [comment]
Speaking of CSS animations, I've heard that Angular is switching to native CSS animations from the Material team.

I was wondering what would be the roadmap of the animations library? Angular animations still seem to be a better option to me because of the leaving animations and the coordination with child animations.

Thanks for your wonderful Angular presentations btw! And thanks for adding Kirill to review group!
I probably wouldn't have gotten into Northeastern this semester without Kirill, and I'm so glad he is part of the Angular team now!

**Comment by thePunderWoman** [comment]
We've now put up [a guide](https://angular.dev/guide/animations/migration) for migrating from `@angular/animations` to native CSS animations, and we would recommend using CSS over the animations package in general. Putting the animations in css and using `@starting-styles` is the proper solution to this problem.
